yb_migrate/src/srcdb/ora2pg_extract_schema.go

```diff
@@ -18,77 +16,52 @@ func ora2pgExtractSchema(source *Source, exportDir string) {
 		}
 
-		fmt.Printf("exporting %10s %5s", exportObject, "")
-
+		fmt.Printf("exporting %10s %5s", objectType, "")
 		go utils.Wait(fmt.Sprintf("%10s\n", "done"), fmt.Sprintf("%10s\n", "error!"))
 
-		exportObjectFileName := utils.GetObjectFileName(schemaDirPath, exportObject)
-		exportObjectDirPath := utils.GetObjectDirPath(schemaDirPath, exportObject)
+		outFile := utils.GetObjectFileName(schemaDirPath, objectType)
+		outDir := utils.GetObjectDirPath(schemaDirPath, objectType)
 
-		var exportSchemaObjectCommand *exec.Cmd
-		if source.DBType == "oracle" {
-			exportSchemaObjectCommand = exec.Command("ora2pg", "-p", "-q", "-t", exportObject, "-o",
-				exportObjectFileName, "-b", exportObjectDirPath, "-c", configFilePath, "--no_header")
-			log.Infof("Executing command: %s", exportSchemaObjectCommand.String())
-		} else if source.DBType == "mysql" {
-			exportSchemaObjectCommand = exec.Command("ora2pg", "-p", "-m", "-q", "-t", exportObject, "-o",
-				exportObjectFileName, "-b", exportObjectDirPath, "-c", configFilePath, "--no_header")
-			log.Infof("Executing command: %s", exportSchemaObjectCommand.String())
+		args := []string{
+			"-p", "-q",
+			"-t", objectType,
+			"-o", outFile,
+			"-b", outDir,
+			"-c", configFilePath,
+			"--no_header",
 		}
+		if source.DBType == "mysql" {
+			args = append(args, "-m")
+		}
+		cmd := exec.Command("ora2pg", args...)
+		log.Infof("Executing command: %s", cmd.String())
 
-		stdout, _ := exportSchemaObjectCommand.StdoutPipe()
-		stderr, _ := exportSchemaObjectCommand.StderrPipe()
-
-		go func() { //command output scanner goroutine
-			outScanner := bufio.NewScanner(stdout)
-			for outScanner.Scan() {
-				line := strings.ToLower(outScanner.Text())
-				if strings.Contains(line, "error") {
-					utils.WaitChannel <- 1 //stop waiting with exit code 1
-					<-utils.WaitChannel
-					log.Infof("ERROR in output scanner goroutine: %s", line)
-					runtime.Goexit()
-				} else {
-					log.Infof("ora2pg STDOUT: %s", outScanner.Text())
-				}
-			}
-		}()
-
-		go func() { //command error scanner goroutine
-			errScanner := bufio.NewScanner(stderr)
-			for errScanner.Scan() {
-				line := strings.ToLower(errScanner.Text())
-				if strings.Contains(line, "error") {
-					utils.WaitChannel <- 1 //stop waiting with exit code 1
-					<-utils.WaitChannel
-					log.Infof("ERROR in error scanner goroutine: %s", line)
-					runtime.Goexit()
-				} else {
-					utils.PrintAndLog("ora2pg STDERR: %s", errScanner.Text())
-				}
-			}
-		}()
+		var outbuf bytes.Buffer
+		var errbuf bytes.Buffer
+		cmd.Stdout = &outbuf
+		cmd.Stderr = &errbuf
 
-		err := exportSchemaObjectCommand.Start()
+		err := cmd.Start()
 		if err != nil {
-			utils.PrintAndLog("Error while starting export: %v", err)
-			exportSchemaObjectCommand.Process.Kill()
-			continue
+			utils.WaitChannel <- 1 //stop waiting with exit code 1
+			<-utils.WaitChannel
+			utils.ErrExit("Failed to initiate %s export: %v\n%s", objectType, err, errbuf.String())
```

Comment:
Have you verified that we don't need to wait/sleep after line 52 and before line 53